adam/experiment/make_integrated_experiment.py

```diff
@@ -0,0 +1,220 @@
+                "experiment": experiment_name_string,
+                "experiment_group_dir": directory_for(experiment_name),
+                "hypothesis_log_dir": directory_for(experiment_name) / "hypotheses",
+                "learner_logging_path": directory_for(experiment_name),
+                "log_learner_state": True,
+                "resume_from_latest_logged_state": True,
+                "object_learner": LEARNER_TO_PARAMS[object_learner_type],
+                "attribute_learner": LEARNER_TO_PARAMS[attribute_learner_type],
+                "relation_learner": LEARNER_TO_PARAMS[relation_learner_type],
+                "load_from_curriculum_repository": curriculum_path,
+            }
+        )
+
+        run_python_on_parameters(
+            experiment_name,
+            log_experiment_script,
+            experiment_params,
+            depends_on=[curriculum_dep],
+            resource_request=SlurmResourceRequest.from_parameters(
+                PURSUIT_RESOURCE_REQUEST
+            )
+            if "pursuit"
+            in [object_learner_type, attribute_learner_type, relation_learner_type]
+            else None,
+            category="pursuit"
+            if "pursuit"
+            in [object_learner_type, attribute_learner_type, relation_learner_type]
+            else "subset",
+        )
+
+    write_workflow_description()
+
+
+EXPERIMENT_NAME_FORMAT = "{curriculum_name}-object_learner:{object_learner}-attribute_learner:{attribute_learner}-relation_learner:{relation_learner}"
+CURRICULUM_NAME_FORMAT = (
+    "noise={noise}-shuffled={shuffled}-attributes={attributes}-relations={relations}"
+)
+
+LEARNER_TO_PARAMS = {
+    "subset": {"learner_type": "subset", "ontology": "integrated_experiment"},
+    "pursuit": {
+        "learner_type": "pursuit",
+        "ontology": "integrated_experiment",
+        "random_seed": 0,
+        "learning_factor": 0.02,
+        "graph_match_confirmation_threshold": 0.9,
+        "lexicon_entry_threshold": 0.7,
+        "smoothing_parameter": 0.001,
+    },
+    "none": {"learner_type": "none"},
+}
+
+CURRICULUM_PARAMS = {
+    "block_multiple_of_same_type": True,
+    "include_targets_in_noise": False,
+    "min_noise_objects": 1,
+    "max_noise_objects": 10,
+    "min_noise_relations": 0,
+    "max_noise_relations": 5,
+    "random_seed": 0,
+    "chooser_seed": 0,
+}
+
+FIXED_PARAMETERS = {
+    "curriculum": "m18-integrated-learners-experiment",
+    "learner": "integrated-learner-params",
+    "post_observer": {
+        "include_acc_observer": False,
+        "include_pr_observer": True,
+        "log_pr": True,
+    },
+    "test_observer": {"accuracy_to_txt": True},
+    "action_learner": {"learner_type": "none"},
+    "include_functional_learner": False,
+    "include_generics_learner": False,
+}
+
+PURSUIT_RESOURCE_REQUEST = Parameters.from_mapping(
+    {
+        "exclude_list": f"saga01,saga02,saga03,saga04,saga05,saga06,saga07,saga08,saga09,saga10,saga11,saga12,saga13,"
```

Comment:
I am not sure if this is the problem, but I get an error when I run the following command:

```
(adam) [jcecil@saga03 adam]$ srun --qos=ephemeral --partition=ephemeral --exclude=saga01,saga02,saga03,saga04,saga05,saga06,saga07,saga08,saga09,saga10,saga11,saga12,saga13,saga14,saga15,saga16,saga17,saga18,saga19,saga20,saga21,saga22,saga23,saga24,saga25,saga26,gaia01,gaia02 --pty bash
srun: error: Unable to allocate resources: Invalid node name specified
```

(i.e. when I use the above exclude list verbatim with SLURM)

This error goes away when I delete saga09 from the list. Does it work if you remove that node from the excludes?

Reply:
@LichMaster98 (forgot to tag you)

Reply:
@spigo900 Thanks for catching this. I've adjusted the resource request accordingly. However, I don't think this could be impacting the jobs I currently see fail because I don't make this resource request for the curriculum builder jobs, only for the experiment jobs which contain a pursuit learner.

Reply:
@LichMaster98 Ah, okay, I'll take another look then.